starters/deno-oak-denodb/src/cache/cache.ts

```diff
@@ -0,0 +1,63 @@
+import { connect, Redis, SetOpts } from '../../deps.ts';
+import { REDIS_CACHE_HOST, REDIS_CACHE_PORT } from '../config/environment.ts';
+import { logger } from '../util/logger.ts';
+
+type GetFreshValue<T> = {
+	(): Promise<T> | T;
+};
+
+const redisHostname = REDIS_CACHE_HOST || 'localhost';
+const redisPort = REDIS_CACHE_PORT ? parseInt(REDIS_CACHE_PORT) : 6379;
+
+export class Cache {
+	redisClient!: Redis;
+	constructor() {
+		this.connectToRedis();
+	}
+
+	private async connectToRedis(): Promise<void> {
+		this.redisClient = await connect({
+			hostname: redisHostname,
+			port: redisPort,
+		});
+		this.redisClient.isConnected && logger.info('Connected to Redis');
+	}
+
+	async readItem<T>({ cacheKey }: { cacheKey: string; ex?: number }): Promise<T | null> {
+		const cacheValue = await this.redisClient.get(cacheKey);
+		if (cacheValue) {
+			return JSON.parse(cacheValue);
+		}
+		return null;
+	}
+
+	async writeItem<T>(
+		{ cacheKey, ex }: { cacheKey: string; ex?: number },
+		callback: GetFreshValue<T>,
```

Comment:
Why do we have a callback here instead of passing just the regular value to be cached?